src/main/java/application/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,190 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get user with login " + login, e);
+        }
+        user.setRoles(getRoles(user.getId()));
+        return Optional.of(user);
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO users(name, login, password) VALUES (?, ?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+            }
+            return insertRoles(user);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create user " + user, e);
+        }
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        User user = null;
+        String query = "SELECT * FROM users INNER JOIN users_roles ON id = user_id "
+                + "WHERE deleted = FALSE AND id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get user with id " + id, e);
+        }
+        user.setRoles(getRoles(user.getId()));
+        return Optional.of(user);
+    }
+
+    @Override
+    public List<User> getAll() {
+        List<User> users = new ArrayList<>();
+        String query = "SELECT * FROM users INNER JOIN users_roles ur ON id = user_id "
+                + "WHERE deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                User user = getUserFromResultSet(resultSet);
+                users.add(user);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get users from DB", e);
+        }
+        for (User user : users) {
+            user.setRoles(getRoles(user.getId()));
+        }
+        return users;
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE users SET name = ?, login = ?, password = ? WHERE id = ?";
```

Comment:
what about deleted user, we don't need to update them

Reply:
done